src/main/msp/msp_serial.c

```diff
@@ -73,52 +75,165 @@ void mspSerialReleasePortIfAllocated(serialPort_t *serialPort)
 
 static bool mspSerialProcessReceivedData(mspPort_t *mspPort, uint8_t c)
 {
```

Comment:
It may be easier (and faster) to compute checksums in blocks when enough data arrive (at end of header / payload)

Reply:
I don't think so. We don't use an optimised function for block checksum calculations - we are calling checksum update function for a bunch of times regardless. Current approach spreads the effort of calculating the checksum over time and in the end gives lower MSP latency.

Reply:
Calling checksum function in loop is easily optimized - compiler can keep data pointer and result in registers. Performance gain is usually non-negligible.

Reply:
There is a difference whether we calculate checum incrementally after each byte arriving slowly or we have to spend time calculating checksum over a complete buffer. While the second may save us a few ticks, it will also increase processing latency. I've been down this road before with OSD in INAV - when bulk updates introduced latency that messed up with calculations that should be isochronous - thus if I have to chose over lowering latency vs lowering CPU load the decision if pretty much obvious.

Reply:
I agree with @digitalentity , latency is more important than raw speed here.